libs/dextool/source/dextool/compilation_db/package.d

```diff
@@ -739,19 +764,31 @@ ParseFlags parseFlag(CompileCommand cmd, const CompileCommandFilter flag_filter,
                 st = State.keep;
             } else if (st == State.pathArgumentToAbsolute) {
                 st = State.keep;
-                auto p = buildNormalizedPath(workdir, arg).absolutePath;
-                rval.put(p);
-                includes.put(p);
+                putNormalizedAbsolute(arg);
             } else if (st == State.priorityKeepNextArg) {
                 st = State.keep;
                 rval.put(arg);
+            } else if (st == State.checkingForQuotation) {
+                if (!path.data.empty) {
+                    path.put(" ");
+                }
+                path.put(arg);
+                if (isEndingWithQuotation(arg)) {
+                    st = State.keep;
+                    putNormalizedAbsolute(path.data.join.strip('"'));
+                    path.clear();
+                }
             } else if (excludeStartWith(arg, flag_filter)) {
                 st = State.skipIfNotFlag;
+            } else if (isSpacedInclude(arg)) {
+                st = State.checkingForQuotation;
+                rval.put("-I");
+                if (!arg[2 .. $].empty) {
```

Comment:
I recommend changing this to `arg.length >= 4`. It namely took me a while to "see" that it is safe to take a slice like this because `isSpacedInclude` is only true if the length is >= 3.